src/multi_agent/core/agent_initializer.py

```
@@ -0,0 +1,93 @@
+    # Initialize provider (auto-detects from model name: claude/gpt/gemini)
+    try:
+        provider = create_provider(model=config.model)
+        logger.info(f"Initialized provider for {agent_name} with model: {config.model}")
+    except KeyError as e:
+        # Missing API key
+        raise APIKeyError(
+            f"Missing API key for model {config.model}. "
+            f"Ensure the appropriate API key is set in environment variables.",
+            details={"model": config.model, "agent": agent_name},
+            cause=e
+        )
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)